src/main/scala/NumberFormat.scala

```diff
@@ -0,0 +1,39 @@
+package CSVValidation
+
+import Errors.NumberFormatError
+import com.ibm.icu.text.DecimalFormat
+
+case class NumberFormat(pattern:Option[String], groupChar: Option[Char] = None, decimalChar: Option[Char] = None) {
+  private var df:DecimalFormat = new DecimalFormat()
+  private val decimalFormatSymbols = df.getDecimalFormatSymbols
+  decimalFormatSymbols.setInfinity("INF")
+  try {
+    decimalChar match {
+      case Some(c) => decimalFormatSymbols.setDecimalSeparator(c)
+      case _ => {}
```

Comment:
Thinking about it, if no decimal char is specified, we should set the decimal separator specifically to '.' - and similarly with the grouping char.

We don't know which locale the user running this script will have, the default locale that the ICU library selects for the user might not have '.' as the decimal char and ',' as the group. So we need to ensure that's the case if we get None for these inputs.

Reply:
Make sure you set a comment explaining that why you're doing those things.

Reply:
done